gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -186,11 +209,11 @@ public static ClientContext create(StubSettings settings) throws IOException {
     if (transportChannelProvider.needsExecutor() && settings.getExecutorProvider() != null) {
       transportChannelProvider = transportChannelProvider.withExecutor(backgroundExecutor);
     }
-    Map<String, String> headers = getHeadersFromSettings(settings);
+
     if (transportChannelProvider.needsHeaders()) {
       transportChannelProvider = transportChannelProvider.withHeaders(headers);
     }
-    if (transportChannelProvider.needsCredentials() && credentials != null) {
+    if (!hasApiKey && transportChannelProvider.needsCredentials() && credentials != null) {
```

Comment:
this additional check seem redundant, any reason for that?

Reply:
It is not redundant. `transportChannelProvider.needsCredentials()` may not check if API key is present.

Reply:
Yes, but you set credentials only if !hasApiKey. Otherwise they would be null.

Reply:
yep, removed `&& credentials != null`